crates/bevy_ptr/src/lib.rs

```diff
@@ -539,7 +539,7 @@ pub const fn dangling_with_align(align: NonZeroUsize) -> NonNull<u8> {
     debug_assert!(align.is_power_of_two(), "Alignment must be power of two.");
     // SAFETY: The pointer will not be null, since it was created
     // from the address of a `NonZero<usize>`.
-    unsafe { NonNull::new_unchecked(align.get() as *mut u8) }
+    unsafe { NonNull::new_unchecked(ptr::null_mut::<u8>().wrapping_add(align.get())) }
```

Comment:
should we leave a comment to switch this to `with_addr` once that stabilizes?

Reply:
The even better alternative would be [`without_provenance_mut`](https://doc.rust-lang.org/stable/std/ptr/fn.without_provenance_mut.html), but that will likely be stabilized even later.